From b05202920a62bec70b70b23a7f126dba91d75771 Mon Sep 17 00:00:00 2001
From: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Date: Tue, 28 Feb 2023 18:27:22 +0100
Subject: [PATCH] arm64: dts: st: add rpmsg i2c bus for stm32mp257f-ev1

Add rpmsg i2c bus support for communication with the Cortex-M33.
Linux drivers can register to this bus to simulate an I2C link over
rpmsg.

Signed-off-by: Arnaud Pouliquen <arnaud.pouliquen@foss.st.com>
Change-Id: Ibd7bdc39526677817a9bb609e08c489deba17b9d
---
 arch/arm64/boot/dts/st/stm32mp257f-ev1.dts | 9 +++++++++
 1 file changed, 9 insertions(+)

--- a/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
+++ b/arch/arm64/boot/dts/st/stm32mp257f-ev1.dts
@@ -193,6 +193,15 @@
 		interrupt-controller;
 		status = "okay";
 	};
+
+	i2c_rpmsg: i2c@2 {
+		compatible = "rpmsg,i2c-controller";
+		reg = <2 0>;
+		rpmsg,dev-id = "rpmsg_i2c";
+		#address-cells = <1>;
+		#size-cells = <0>;
+		status = "okay";
+	};
 };
 
 &ommanager {
